src/mqtt_pal.c

```diff
@@ -106,6 +106,83 @@ ssize_t mqtt_pal_recvall(mqtt_pal_socket_handle fd, void* buf, size_t bufsz, int
 
 #endif
 
+#elif defined(_MSC_VER)
```

Comment:
@LiamBindle imo it would make sense to merge the `unix` and `!unix` case, is that ok with you?

Reply:
I'd prefer to keep them seperate. The defaults in `mqtt_pal.c` are more meant to be examples of how to port MQTT-C to different systems.

Reply:
Instead of modifying this file, I should add `mqtt_pal_msvc.c` ?

Reply:
What you've done in `mqtt_pal.c` looks good to me.

Reply:
The BIO case is now shared between win32 and unix.
When the casts are added there, it can compile on both.
The unix case now differs from the win32 case in the `errno` check, which is done with `WSAGetLastError()` on win32